### PR TITLE
Migrate to sbt 1.0.0-M6

### PR DESCRIPTION
**Pull request by jvican**
Dogfooding ourselves.

### CODE REVIEW

**Comment by jvican** [comment]
Things blocking this PR:
  
- [x] sbt-jmh
- [x] neo-sbt-scalafmt

**Comment by jvican** [comment]
This PR depends on https://github.com/sbt/zinc/pull/315. It cannot be merged after that has landed. CI is not passing because we still require the aforementioned sbt plugins. I published locally for both and worked.
  
We need to wait to merge this until we have the scripted and test log improvements in a release.

**Comment by eed3si9n** [comment]
I would like to get emulated monorepo working on my machine again, and I am currently blocked on Zinc not having the same sbt version.

**Comment by jvican** [comment]
@eed3si9n Honestly, I'm not happy to merge sbt 1.0.x support until we don't have the logging improvements that make test logs intelligible. Is it possible to release a bincompat M6-2 and we depend on it?

**Comment by jvican** [comment]
By the way, PR has been updated and sbt 1.0 works. Old scalafmt plugin has been removed in favor of the thoroughly-tested sbt-neo-scalafmt.

File: build.sbt [hunk 1]
```diff
@@ -56,7 +55,7 @@ def baseSettings: Seq[Setting[_]] =
 //   minimalSettings ++ baseScalacOptions ++ Licensed.settings ++ Formatting.settings
 
 def addBaseSettingsAndTestDeps(p: Project): Project =
-  p.settings(baseSettings).configure(addTestDependencies)
+  p.settings(baseSettings).configure(addTestDependencies _)
```
Comment (dwijnand) [inline build.sbt]: By-the-by, I'm pretty sure the underscores aren't required here and throughout. I think I know what happened: if you make a mistake, like forget a comma, the scala compiler is eager to tell how you should be explicitly eta-expanding these, but the error is actually elsewhere (i.e the missing comma).

Comment (jvican) [reply build.sbt]: Thanks for the heads up Dale, I didn't know this and yeah, it's probably what happened. I've just updated the PR. It's a shame scalac behaves this way, hope this changes.

**Comment by eed3si9n** [comment]
```
[error] (zinc/compile:scalafmt::test) java.lang.NoSuchMethodError: com.lucidchart.scalafmt.api.Scalafmtter.formatter(Lcom/lucidchart/scalafmt/api/Dialect;)Ljava/util/function/Function;
101s
289
[error] (zinc/compile:scalafmt::sources) java.lang.NoSuchMethodError: com.lucidchart.scalafmt.api.Scalafmtter.includeFile(Ljava/nio/file/Path;)Z
101s
290
[error] (zinc/compile:scalafmt::sources) java.lang.NoSuchMethodError: com.lucidchart.scalafmt.api.Scalafmtter.includeFile(Ljava/nio/file/Path;)Z
101s
291
[error] (zincCompile/compile:scalafmt::test) java.lang.NoSuchMethodError: 
```

Could we first pick the same `addSbtPlugin("com.lucidchart" % "sbt-scalafmt" % "1.3")` that's been working for other modules, and add improvements in a separate PR?

**Comment by jvican** [comment]
Closing in favor of #323.

**Pull request closed without merging**